r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+make life easier. Once a project is opened within RStudio the following actions
+are taken:
+
+- A new R session (process) is started.
+- The current working directory is set to the project directory.
+- RStudio project options are loaded.
+
+The README in each folder explains a bit about what should be placed there. But
+briefly:
+
+1. Documents like manuscripts, abstracts, and exploration type documents should 
+be put in the `doc/` directory (including [R Markdown][r-markdown] files which
+we will cover [later](#r-markdown)).
+1. Data, raw data, and metadata should be in either the `data/` directory or
+in `data-raw/` for the raw data.
+1. All R files and code should be in the `R/` directory.
+1. Name all new files to reflect their content or function. Follow the tidyverse 
+[style guide for file naming][r-style-guide-files].
+
+Note the `DESCRIPTION` file. This is used as metadata about the project and is 
+useful when working on R projects. For any project, it is **highly recommended**
+to use [version control]. We'll be covering version control in more detail
+[later](#version-control) in the course.
+
+TODO: Make sure the internal reference links are correct.
+
+### Exercise: Better file naming
+
+Look at the list of file names below. Which file names are good names and which
+shouldn't you use? 
+
+```
+fit models.R
+fit-models.R
+foo.r
+stuff.r
+get_data.R
+Manuscript version 10.docx
+manuscript.docx
+new version of analysis.R
+trying.something.here.R
+plotting-regression.R
+utility_functions.R
+code.R
+```
+
+### Should you keep your data under version control?
+
+We have a `data/` folder for a reason. But you might not want to keep the data
+under version control, for several reasons:
+
+1. It's a large dataset (tens or more of Mb file size)
+2. There are sensitive or personal information in the data
+
+As a rule of thumb, if you can send the data by an email attachment, you could
+probably put it into Git. Unless there is sensitive or personal data, then don't.
+If it isn't kept under version control, make sure you include a reference to how
+or where you got the data, either as an R script showing the code you used to 
+import/clean/download it or described in the `README.md` file.
+
+## Reusability
+
+Part of reproducibility is also making sure your scripts and file organization is
+"reusable" meaning that others (or yourself) can run it again. So, for instance,
+making sure to use "relative file paths" compared to "absolute file paths"
+(we'll cover these in a bit). Or indicating which other R packages your code
+depends on. So here we'll cover how to make sure your scripts and project files
+are reusable.
+
+### Keeping a "clean slate"
+
+When you finish writing your R code for the day and close the session, you probably
+will be asked about saving your session. What this does is everything kept in the
+"Environment" (e.g. all objects, functions, or datasets you created and used
```

Comment:
Use `[environment](glossary.html#environment)` to link to a glossary definition of a term like this the first time it's used rather than put it in quotes.